quaint/src/connector/postgres/native/cache.rs

```diff
@@ -0,0 +1,388 @@
+    async fn noop_cache_returns_new_queries_every_time() {
+        run_with_client(|client| async move {
+            let cache = NoOpCache;
+            let sql = "SELECT $1";
+            let types = [Type::INT4];
+
+            let stmt1 = cache.get_query(&client, sql, &types).await.unwrap();
+            let stmt2 = cache.get_query(&client, sql, &types).await.unwrap();
+            assert_ne!(stmt1.name(), stmt2.name());
+        })
+        .await;
+    }
+
+    #[tokio::test]
+    async fn noop_cache_returns_new_statements_every_time() {
+        run_with_client(|client| async move {
+            let cache = NoOpCache;
+            let sql = "SELECT $1";
+            let types = [Type::INT4];
+
+            let stmt1 = cache.get_statement(&client, sql, &types).await.unwrap();
+            let stmt2 = cache.get_statement(&client, sql, &types).await.unwrap();
+            assert_ne!(stmt1.name(), stmt2.name());
+        })
+        .await;
+    }
+
+    #[tokio::test]
+    async fn prepared_statement_lru_cache_reuses_queries_within_capacity() {
+        run_with_client(|client| async move {
+            let cache = PreparedStatementLruCache::with_capacity(1);
+            let sql = "SELECT $1";
+            let types = [Type::INT4];
+
+            let stmt1 = cache.get_query(&client, sql, &types).await.unwrap();
+            let stmt2 = cache.get_query(&client, sql, &types).await.unwrap();
+            assert_eq!(stmt1.name(), stmt2.name());
+
+            // replace our cached statement with a new one going over the capacity
+            cache.get_query(&client, sql, &[Type::INT8]).await.unwrap();
+
+            // the old statement should be evicted from the cache
+            let stmt3 = cache.get_query(&client, sql, &types).await.unwrap();
+            assert_ne!(stmt1.name(), stmt3.name());
+        })
+        .await;
+    }
+
+    #[tokio::test]
+    async fn prepared_statement_lru_cache_reuses_statements_within_capacity() {
+        run_with_client(|client| async move {
+            let cache = PreparedStatementLruCache::with_capacity(1);
+            let sql = "SELECT $1";
+            let types = [Type::INT4];
+
+            let stmt1 = cache.get_statement(&client, sql, &types).await.unwrap();
+            let stmt2 = cache.get_statement(&client, sql, &types).await.unwrap();
+            assert_eq!(stmt1.name(), stmt2.name());
+
+            // replace our cached statement with a new one going over the capacity
+            cache.get_statement(&client, sql, &[Type::INT8]).await.unwrap();
+
+            // the old statement should be evicted from the cache
+            let stmt3 = cache.get_statement(&client, sql, &types).await.unwrap();
+            assert_ne!(stmt1.name(), stmt3.name());
+        })
+        .await;
+    }
+
+    #[tokio::test]
+    async fn tracing_lru_cache_reuses_queries_within_capacity() {
+        run_with_client(|client| async move {
+            let cache = TracingLruCache::with_capacity(1);
```

Comment:
Let's expand this test case with a higher capacity, e.g., 3.

Reply:
done